### DISCUSSION TITLE
502 Bad Gateway after update 1.4.4 to 1.5.0

### DISCUSSION THREAD

**Discussion by Burschi500** [jonaswinkler/paperless-ng#1331] [Support]
Good morning,

after updating to 1.5.0 (bare metal) and copying over the paperless.conf and adjusting ownership i get a 502 when accessing paperless. I use nginx as proxy, but only for internal use.

What did i do wrong during updating?

**Comment by TobenderZephyr** [comment]
Hi Burschi

Error 502/503 Bad Gateway usually appears only on the proxy itself who can't reach its backend.
Since it doesn't seem to be running, with that little information at hand it is next to impossible to help.

Does the process itself run and do you have any logfiles in your paperless directory?

**Reply by Burschi500** [reply]
Yes, too less info, I was hoping that this was maybe a common error with an easy answer. 

This is from the nginx error.log
`2021/09/21 11:15:59 [error] 200#200: *166 connect() failed (111: Connection refused) while connecting to upstream, client: 192.168.20.254, server: , request: "GET /accounts/login/?next=/ HTTP/1.1", upstream: "http://[::1]:8000/accounts/login/?next=/", host: "192.168.20.236"`
Upstream seems to be ipv6 instead of ipv4 127.0.0.1
Nginx itself doesnt seem to be the problem, since there was no change. When i copy the old files back everything works as expected.

**Reply by TobenderZephyr** [reply]
For localhost, it shouldn't really matter if nginx connects via IPv4 or IPv6.
connection refused shows that your machine does not accept connections to port 8000.

I was hoping to see recent logs of paperless-ng 1.5.0. You find those inside `PAPERLESS_LOGGING_DIR`
usually defaults to `<Install directory>/data/logs`
If you configured paperless-ng to take advantage of systemd for automatic startup, you could also check `journalctl -xn`

Were you actually running the required steps for updating paperless-ng?

```pip install requirements.txt```
and
```
cd src
python3 manage.py migrate
```

**Reply by Burschi500** [reply]
Did this. It seems gunicorn did not start up, because the files are missing under <Install directory>/.local/bin I only have the lib directory. Using 'sudo -Hu paperless python3 manage.py runserver' everything works fine.
Question is why the files under .local are missing and what to do to get them installed properly.
